specs/IsNonClientRegionSupportEnabled.md

```diff
@@ -0,0 +1,141 @@
+
+# Description
+`IsNonClientRegionSupportEnabled` defaults to `FALSE`. Disabling/Enabling 
+`IsNonClientRegionSupportEnabled` takes effect after the next navigation. Currently, draggable 
+regions is the only non-client experience we have implemented. Eventually, this setting will 
+expand to enable other non-client functionality, such as resize and caption controls. 
+
+When the setting is set to `TRUE`, then the following non-client region support for the top 
+level document will be enabled:  
+* Web pages will be able to use the `app-region` CSS style. 
+* Draggable Regions declared with the CSS style `app-region: drag` will support title bar 
+functionality, for example, the dragging of the app window, the title bar context menu upon 
+right click, and the maximizing/restoring of the window size upon double click of the html 
+element. 
+
+When set to `FALSE`, then all non-client region support will be disabled.  
+* Web pages will not be able to use the `app-region` CSS style. 
+
+# Examples
+```cpp 
+ScenarioNonClientRegionSupport::ScenarioNonClientRegionSupport(AppWindow* appWindow)
+    : m_appWindow(appWindow), m_webView(appWindow->GetWebView())
+{
+    
+    CHECK_FAILURE(m_webView->add_NavigationStarting(
+        Callback<ICoreWebView2NavigationStartingEventHandler>(
+            [this](
+                ICoreWebView2* sender,
+                ICoreWebView2NavigationStartingEventArgs* args) -> HRESULT
+            {
+                static const PCWSTR url_compare_example = L"www.microsoft.com";
+                wil::unique_cotaskmem_string uri;
+                CHECK_FAILURE(args->get_Uri(&uri));
+                wil::unique_bstr domain = GetDomainOfUri(uri.get());
+                
+                wil::com_ptr<ICoreWebView2Settings> m_settings;
+                CHECK_FAILURE(m_webView->get_Settings(&m_settings));
+                wil::com_ptr<ICoreWebView2Settings12> coreWebView2Settings12;
+                coreWebView2Settings12 = m_settings.try_query<ICoreWebView2Settings12();
+                CHECK_FEATURE_RETURN(coreWebView2Settings12);
+                
+                BOOL enabled;
+                CHECK_FAILURE(coreWebView2Settings12->get_IsNonClientRegionSupportEnabled(&enabled));
+
+                if (wcscmp(domain.get(), url_compare_example) == 0 && !enabled)
+                {
+                    CHECK_FAILURE(
+                        coreWebView2Settings12->put_IsNonClientRegionSupportEnabled(TRUE));
+                }
+                else if (wcscmp(domain.get(), url_compare_example) != 0 && enabled)
+                {
+                    CHECK_FAILURE(
+                        coreWebView2Settings12->put_IsNonClientRegionSupportEnabled(FALSE));
+                }
+                return S_OK;
+            })
+            .Get(),
+        &m_navigationStartingToken));
+}
+```
+
+```c#
+private WebView2 webView;
+webView.CoreWebView2.NavigationStarting += SetNonClientRegionSupport;
```

Comment:
This part looks sort of wrong. Line 76 has a class member definition, and then following that on line 77 you have code that can't run in a class definition. Maybe add part of the init method that is calling NavigationStarting += ...